### PR TITLE
chore: updates Smart Card IdP documentation in the SIW

### PR DESCRIPTION
**Pull request by gowthamidommety-okta**
Resolves: OKTA-295394

## Description:

Updates X509 documentation in the SIW

Screen Shot 2020-06-29 at 2 03 48 PM

Screen Shot 2020-06-29 at 2 03 26 PM

Note clicking on the EA Feature link takes us to https://developer.okta.com/docs/reference/releases-at-okta/#early-access-ea

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-295394](https://oktainc.atlassian.net/browse/OKTA-295394)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1120,6 +1121,27 @@ Options for the [OpenID Connect](http://developer.okta.com/docs/api/resources/oi
...
+
+- `className` - Optional class that can be added to the Smart Card IdP button.
+
+- `isCustomDomain` - Optional boolean that is passed to the certAuthUrl endpoint to indicate whether the request comes from a custom domain.
+
+    ```javascript
+    piv: {
+      text: 'Authenticate with a Smart Card',
+      className: 'custom-style',
+      certAuthUrl: 'https://${yourOktaDomain}.okta.com/login/cert',
```
Comment (abdullahchougle-okta) [inline README.md:1140]: Shouldn't this be `https://${yourOktaDomain}.okta.com/api/internal/v1/authn/cert`? `https://${yourOktaDomain}.okta.com/login/cert` will lead to the JSP instead.

@nikolaystoilov-okta will it be a problem if we reference an internal API here?

File: README.md [hunk 2]
```diff
@@ -1120,6 +1121,27 @@ Options for the [OpenID Connect](http://developer.okta.com/docs/api/resources/oi
     }
     ```
 
+### Smart Card IdP
+
+Options for authentication with a Smart Card `X509` type IdP.
```
Comment (mauriciocastillosilva-okta) [inline README.md:1126]: nit: "**Settings** for authentication with a Smart Card `X509` type IdP."

File: README.md [hunk 3]
```diff
@@ -1120,6 +1121,27 @@ Options for the [OpenID Connect](http://developer.okta.com/docs/api/resources/oi
...
+
+- `certAuthUrl` - Required url endpoint that requests the browser for a certificate.
```
Comment (mauriciocastillosilva-okta) [inline README.md:1128]: we can probably omit "Required" and just add "Optional" in the ones we need. This is similar to other configs we have. 
nit: we should uppercase URL
```
`certAuthUrl` - URL endpoint that requests the browser for a certificate.
`text` *(optional)* - 
`className ` *(optional)* - 
etc
```
same comment for the other optional params

File: README.md [hunk 4]
```diff
@@ -1120,6 +1121,27 @@ Options for the [OpenID Connect](http://developer.okta.com/docs/api/resources/oi
...
+
+- `text` - Optional string that can be set as the Smart Card IdP button text. By default, this value will be "Sign in with PIV / CAC card".
```
Comment (mauriciocastillosilva-okta) [inline README.md:1130]: what about
```
- `text` *(optional)* - Label for the Smart Card IdP button. By default, this value will be "Sign in with PIV / CAC card".
```

File: README.md (hunk 1 above)
Comment (gowthamidommety-okta) [reply README.md:1140]: Yea, this was just an example.. I wasn't sure if we could reference an internal api here. Will wait for Nikolay's input

Comment (mauriciocastillosilva-okta) [reply README.md:1140]: +1 on leaving it as is. This is just an example and we shouldn't make any reference to internal APIs in our doc

Comment (abdullahchougle-okta) [reply README.md:1140]: ok, but it's a confusing example since `/login/cert` won't ever work...

Comment (mauriciocastillosilva-okta) [reply README.md:1140]: its a good point... but we don't have a public endpoint that will work, correct? We can make it more obvious that this is just an example with something like `/your/cert/validation/endpoint`?

Comment (abdullahchougle-okta) [reply README.md:1140]: That might be an OK middle ground.

I assume that the customer will either need to contact Okta or see how the default SIW works to figure out the real URL.

**Comment by haishengwu-okta** [comment]
looks like `certAuthUrl` is required? https://github.com/okta/okta-signin-widget/blob/master/src/models/Settings.js#L293

**Comment by magizh-okta** [comment]
@gowthamidommety-okta  just double checking if it works with custom signon/url pages where okta still hosts. If not I dont see a point in documenting since it cannot be used by customers anyway.

**Comment by gowthamidommety-okta** [comment]
> looks like `certAuthUrl` is required? https://github.com/okta/okta-signin-widget/blob/master/src/models/Settings.js#L293

@haishengwu-okta We added a comment here, noting that this feature would only be available to okta hosted SIW. In those cases, the certAuthUrl is passed by default from okta.

**Comment by gowthamidommety-okta** [comment]
> @gowthamidommety-okta just double checking if it works with custom signon/url pages where okta still hosts. If not I dont see a point in documenting since it cannot be used by customers anyway.

@magizh-okta @abdullahchougle-okta For a SIW that okta hosts (okta default hosted / okta custom hosted) , the PIV configuration will work by default. This documentation is only so customers can know how to change the piv button text and add a custom piv button className if needed.

**Comment by gowthamidommety-okta** [comment]
Cannot merge this in. Created a new PR https://github.com/okta/okta-signin-widget/pull/1280

**Pull request closed without merging**